code_examples/core_features/did/01_light_did_simple.ts

```diff
@@ -1,21 +1,13 @@
-import type { Keyring } from '@polkadot/api'
-
-import { randomAsU8a } from '@polkadot/util-crypto'
-
 import * as Kilt from '@kiltprotocol/sdk-js'
 
-export function createSimpleLightDid(
-  keyring: Keyring,
-  authenticationSeed: Uint8Array = randomAsU8a(32)
-): Kilt.DidDocument {
-  // Ask the keyring to generate a new keypair to use for authentication with the generated seed.
-  const authKey = keyring.addFromSeed(
-    authenticationSeed
-  ) as Kilt.NewLightDidVerificationKey
-
+export function createSimpleLightDid({
+  authentication
+}: {
+  authentication: Kilt.KiltKeyringPair
```

Comment:
Would it work like this? We could skip the type casting below in this case.
```suggestion
  authentication: Kilt.NewLightDidVerificationKey
```

Reply:
But if the function returns a DID key, we can't use it for a sign callback, right? The key does not have a `.sign()` function.

Reply:
I meant this part:
```ts
  const lightDID = Kilt.Did.createLightDidDocument({
    authentication: [authentication as Kilt.NewLightDidVerificationKey]
  })

Reply:
Do you think this is a good fix? https://github.com/KILTprotocol/docs/pull/160/commits/e338aebd35982e00a49aad77f30e0446e9faeb78

I wanted to keep parameters for light and full DID creation the same, but that means that for a full DID creation also the sign callback must be given now, as the key cannot be used to sign the payload directly. What do you think?